ccb.go

```diff
@@ -0,0 +1,299 @@
+package main
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/kataras/iris"
+)
+
+// responseHandler is a function that handles the response from CCB.
+// depending on the response from CCB, we may want to reformat the structs before passing them back to the user.
+type responseHandler func(ctx iris.Context, response CCBResponse)
+
+// makeCCBRequest can currently handle
+func makeCCBRequest(ctx iris.Context, url string, method string, handler responseHandler) {
+	// build http request
+	client := &http.Client{}
+	req, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error: Unable to build request to database. " + err.Error())
+		return
+	}
+	req.SetBasicAuth(os.Getenv("CCB_USERNAME"), os.Getenv("CCB_PASSWORD"))
+
+	// send request
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error: Unable to send request to database. " + err.Error())
+		return
+	}
+
+	// TODO: Error handling for error responses from CCB
+	defer resp.Body.Close()
+
+	var data CCBResponse
+
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	err = xml.Unmarshal(respBody, &data)
+	if err != nil {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error unmarshalling response from database")
+		return
+	}
+
+	handler(ctx, data)
+}
+
+func whoIsResponseHandler(ctx iris.Context, resp CCBResponse) {
+	jsonResponse, err := json.Marshal(resp)
+	if nil != err {
+		fmt.Println(err) // TODO: this should be logged
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString("Error marshalling to JSON")
+		return
+	}
+
+	// write back the body as JSON
+	ctx.Write([]byte(jsonResponse))
+}
+
+// formResponseHandler changes the structure of the CCB response data to a more readable structure
+// before passing the JSON back to the user.
+func formResponseHandler(ctx iris.Context, resp CCBResponse) {
+	// fill in count field from CCBResponse to a FormResponses struct
+	// create variables for iterator to fill in.
+	var formResponses FormResponses
+	formResponses.Count = resp.Response.FormResponses.Count
+	var jsonResponse []byte
+	var err error
+
+	// if there are more than 0 form responses returned, fill in the FormResponses struct
+	if formResponses.Count != "0" {
```

Comment:
Count is a string? that's interesting. Is it possible to serialize this as an int instead?

Reply:
Good point, yeah I think we can unmarshal the xml field to an int